src/traces/surface/convert.js

```diff
@@ -342,46 +373,87 @@ proto.update = function(data) {
      * which is the transpose of 'gl-surface-plot'.
      */
 
-    var xcalendar = data.xcalendar,
-        ycalendar = data.ycalendar,
-        zcalendar = data.zcalendar;
+    var i, j, k, v;
+    var rawCoords = [];
+    for(i = 0; i < 3; i++) {
+        rawCoords[i] = [];
+        for(j = 0; j < xlen; j++) {
+            rawCoords[i][j] = [];
+            /*
+            for(k = 0; k < ylen; k++) {
+                rawCoords[i][j][k] = undefined;
+            }
+            */
+        }
+    }
+
+    // coords x, y & z
+    for(j = 0; j < xlen; j++) {
+        for(k = 0; k < ylen; k++) {
+            rawCoords[0][j][k] = this.getXat(j, k, data.xcalendar, sceneLayout.xaxis);
+            rawCoords[1][j][k] = this.getYat(j, k, data.ycalendar, sceneLayout.yaxis);
+            rawCoords[2][j][k] = this.getZat(j, k, data.zcalendar, sceneLayout.zaxis);
+        }
+    }
 
-    fill(coords[2], function(row, col) {
-        return zaxis.d2l(z[col][row], 0, zcalendar) * scaleFactor[2];
-    });
+    // Note: log axes are not defined in surfaces yet.
+    // but they could be defined here...
```

Comment:
They seem to work ok in `latest`: https://codepen.io/etpinard/pen/MzZLap?editors=0010

Reply:
A [demo](https://codepen.io/MojtabaSamimi/pen/GwPzOR?editors=0010) displaying the issue with logs in z direction. Noting that similar issue may also happen in x and y directions when using log.

Reply:
This log issue is now reported here https://github.com/plotly/plotly.js/issues/3312